### PR TITLE
Fixes #550 Mouse position plugin improvements

### LINKED ISSUE #550: MousePosition plugin improvements
* Allow working on click instead of onMouseMove
 * Add optional "Copy coordinates to Clipboard" functionality

### PR DESCRIPTION
**Pull request by nmco**
Associated issue: https://github.com/geosolutions-it/MapStore2/issues/550

### CODE REVIEW

File: web/client/plugins/MousePosition.jsx [hunk 1]
```diff
@@ -39,7 +53,9 @@ const MousePositionButton = connect((state) => ({
     onClick: changeMousePositionState
 })(require('../components/buttons/ToggleButton'));
 
-const MousePositionPlugin = connect(selector)(require('../components/mapcontrols/mouseposition/MousePosition'));
+const MousePositionPlugin = connect(selector, {
+    copyToClipboardEnabled: true
```
Comment (mbarto) [inline web/client/plugins/MousePosition.jsx]: remove this, we cannot disable it in any way otherwise

File: web/client/plugins/MousePosition.jsx [hunk 2]
```diff
@@ -17,10 +17,24 @@ const {changeMousePositionCrs, changeMousePositionState} = require('../actions/m
 
 const selector = createSelector([
     mapSelector || {},
-    (state) => state.mousePosition || {}
-], (map, mousePosition) => ({
+    (state) => state.mousePosition || {},
+    (state) => {
```
Comment (mbarto) [inline web/client/plugins/MousePosition.jsx]: I don't think this is correct. What we need to make the component work on click is another flag in state.mousePosition (similar to showCenter, something like showClick) and use that to choose between state.mousePosition.position and state.mapInfo.clickPoint

File: web/client/components/mapcontrols/mouseposition/MousePosition.jsx [hunk 3]
```diff
@@ -7,6 +7,10 @@
  */
 var React = require('react');
 var proj4js = require('proj4');
+var BootstrapReact = require('react-bootstrap');
```
Comment (mbarto) [inline web/client/components/mapcontrols/mouseposition/MousePosition.jsx]: const {Glyphicon} = require('react-bootstrap');

File: web/client/components/mapcontrols/mouseposition/MousePosition.jsx [hunk 4]
```diff
@@ -31,15 +39,21 @@ let MousePosition = React.createClass({
             enabled: true,
             degreesTemplate: MousePositionLabelDMS,
             projectedTemplate: MousePositionLabelYX,
-            style: {}
+            style: {},
+            copyToClipboardEnabled: false,
+            glyphicon: "glyph glyphicon-paste",
```
Comment (mbarto) [inline web/client/components/mapcontrols/mouseposition/MousePosition.jsx]: glyphicon: "paste"

File: web/client/product/stores/store.js [hunk 5]
```diff
@@ -54,6 +54,9 @@ module.exports = (plugins) => {
             newState = assign(newState, mobileOverride);
         }
 
+        // mouse position will only be show on map click
+        assign(newState, assign(newState.mousePosition || {}, {showOnClick: true}));
```
Comment (mbarto) [inline web/client/product/stores/store.js]: We do not want this by default. Remove

**Comment by nmco** [comment]
Updated, deactivating onClick and the clipboard copy.

**Pull request merged**